packages/discovery-provider/src/tasks/calculate_trending_challenges.py

```
@@ -85,20 +88,43 @@ def enqueue_trending_challenges(
     )
     update_start = time.time()
     with challenge_bus.use_scoped_dispatch_queue():
-        latest_blocknumber = get_latest_blocknumber_via_redis(session, redis)
+        # subtract final poa block because db is final_poa_block + latest_acdc_block
+        latest_blocknumber = get_latest_blocknumber(session)
         if latest_blocknumber is None:
             logger.error(
                 "calculate_trending_challenges.py | Unable to get latest block number"
             )
             return
 
-        # subtract final poa block because db is final_poa_block + latest_acdc_block
+        latest_block_datetime = None
+        if is_indexing_core_em():
+            core = get_core_instance()
+            node_info = core.get_node_info()
+            core_chain_id = node_info.chainid
+
+            latest_indexed_block: Optional[CoreIndexedBlocks] = (
+                session.query(CoreIndexedBlocks)
+                .filter(CoreIndexedBlocks.chain_id == core_chain_id)
+                .order_by(CoreIndexedBlocks.height.desc())
+                .first()
+            )
 
-        latest_block_datetime = datetime.fromtimestamp(
-            web3.eth.get_block(latest_blocknumber - helpers.get_final_poa_block())[
-                "timestamp"
-            ]
-        )
+            if latest_indexed_block:
+                block = core.get_block(int(latest_indexed_block.height))
+                if block:
+                    latest_block_datetime = block.timestamp.ToDatetime()
+        else:
+            latest_block_datetime = datetime.fromtimestamp(
+                web3.eth.get_block(latest_blocknumber - helpers.get_final_poa_block())[
+                    "timestamp"
+                ]
+            )
+
+        if latest_block_datetime is None:
+            logger.error(
+                "calculate_trending_challenges.py | Unable to get latest block time"
+            )
+            return
```

maybe instead of just returning here, we should throw all the way out to the top level. stuff's wack if we get here

---

I think I was following the earlier pattern where if the block was not found then it also returned None. Should we raise an error there too? An indexed block should always return I'd think